### PR TITLE
Update parseScmUrl() to take in scmUri and scmRepo

### PR DESCRIPTION
**Pull request by minz1027**
- Change `parseScmUrl()` to `parseScmUri()`
- Now the method takes in `scmUri`, `scmRepo.name`  and `scmRepo.branch`

screen shot 2016-10-19 at 8 08 20 pm

### CODE REVIEW

File: launch.go [hunk 1]
```diff
@@ -180,7 +176,7 @@ func launch(api screwdriver.API, buildID, rootDir, emitterPath string) error {
 		return fmt.Errorf("fetching Pipeline ID %q: %v", j.PipelineID, err)
 	}
 
-	scm, err := parseScmURL(p.ScmURL)
+	scm, err := parseScmUri(p.ScmUri, p.ScmRepo.Name, p.ScmRepo.Branch)
```
Comment (d2lam) [inline launch.go]: Don't think we need to pass in branch, it could be `uri[2]`

Comment (minz1027) [inline launch.go]: Fixed

File: launch.go [hunk 2]
```diff
@@ -54,27 +54,23 @@ type scmPath struct {
...
 	return fmt.Sprintf("https://%s/%s/%s#%s", s.Host, s.Org, s.Repo, s.Branch)
 }
 
-// e.g. "git@github.com:screwdriver-cd/launch.git#master"
-func parseScmURL(url string) (scmPath, error) {
-	r := regexp.MustCompile("^git@(.*):(.*)/(.*?)(?:\\.git)?#(.*)$")
-	matched := r.FindStringSubmatch(url)
-	if matched == nil || len(matched) != 5 {
-		return scmPath{}, fmt.Errorf("unable to parse SCM URL %v, match: %q", url, matched)
+func parseScmUri(scmUri, scmName, scmBranch string) (scmPath, error) {
```
Comment (tkyi) [inline launch.go]: We were worried the `url` would be different for github vs some other scm and thought using the scmUri, name, and branch would be a more reliable choice

File: screwdriver/screwdriver.go [hunk 3]
```diff
@@ -96,7 +96,13 @@ type Validator struct {
 // Pipeline is a Screwdriver Pipeline definition.
 type Pipeline struct {
 	ID     string `json:"id"`
-	ScmURL string `json:"scmUrl"`
+	ScmRepo ScmRepo `json:"scmRepo"`
+	ScmUri string `json:"scmUri"`
+}
+
+// ScmRepo is an object inside a Pipeline
```
Comment (jer) [inline screwdriver/screwdriver.go]: Not really. It's more of a struct containing a Name than an object that is somewhere. But what is an SCM Repo?

Comment (minz1027) [inline screwdriver/screwdriver.go]: Fixed!
The new API will have a `scmRepo` object inside the pipeline.
https://beta.api.screwdriver.cd/v4/pipelines

**Pull request merged**